CRM/Twingle/Upgrader.php

```diff
@@ -84,4 +84,29 @@ public function upgrade_5140() {
 
     return TRUE;
   }
+
+  /**
+   * Add the "generic" payment instrument with default values to all profiles
+   * to avoid "Payment method could not be matched to existing payment
+   * instrument." error.
+   *
+   * @return TRUE on success
+   * @throws Exception
+   */
+  public function upgrade_5150() {
+    $this->ctx->log->info('Add "generic" payment instrument to all profiles.');
+
+    $profiles = CRM_Twingle_Profile::getProfiles();
+    if ($profiles) {
+      foreach ($profiles as $profile) {
+        if (!$profile->getAttribute('pi_generic', False)) {
+          $profile->setAttribute('pi_generic', 1);
+          $profile->setAttribute('pi_generic_status', 1);
```

Comment:
This does not match the default value set for this payment instrument in `\CRM_Twingle_Profile::createDefaultProfile()` and I'm not sure we really want a default here, since this option seems to be highly custom by definition.

I'd just leave the upgrade function out, tbh. :shrug:

Reply:
Hard to say what would create more confusion: not setting a default which would lead to the "Payment method could not be matched to existing payment" error (this happened to me when I was testing this PR), or creating posts with an incorrect status without noticing. In the end, it's your choice.

Reply:
By the way, I think that the `NULL` default values do not really serve their purpose. To create a new profile, you always have to specify an XCM profile there and save it before you can use it. The moment you save it, the first option of each selection field for the profile is selected. So the "real" default values are the first options of each selection field.

Reply:
Sure, the profile needs configuration, so there are no "factory" defaults that make sense. Saving a form implies user interaction and thus the user's responsibility. I'd just add a warning to the release notes or a post upgrade message saying that people should review their profile configuration if they use the generic payment method.

Seeing an error at least makes it visible to the user that there's something wrong. Silently setting arbitrary "default" values *might* create inconsistent data, which *might* be even worse.

Reply:
So at this point @mariav0 would just have to `git revert f7612b3e83aeb21ab19d2ed855c54935a1ce1f48` and the PR would be fine to merge, right?